src/features/presence/PresenceCollaborators.js

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import Link from '../../components/link/LinkBase';
+import messages from './messages';
+import PresenceAvatar from './PresenceAvatar';
+import { determineInteractionMessage } from './utils/presenceUtils';
+
+import './PresenceCollaborators.scss';
+
+class PresenceCollaborators extends React.Component<Props> {
+    static propTypes = {
+        /* Custom style override for the avatar */
+        avatarStyle: PropTypes.object,
+        collaborators: PropTypes.arrayOf(
+            PropTypes.shape({
+                /** Url to avatar image. If passed in, component will render the avatar image instead of the initials */
+                avatarUrl: PropTypes.string,
+                /** Users id */
+                id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+                isActive: PropTypes.bool,
+                /** Unix timestamp of when the user last interacted with the document */
+                interactedAt: PropTypes.number,
+                /** The type of interaction by the user */
+                interactionType: PropTypes.string,
+                /** User's full name */
+                name: PropTypes.string.isRequired,
+                /** Custom Profile URL */
+                profileUrl: PropTypes.string,
+            }),
+        ).isRequired,
+        /* Intl object */
+        intl: PropTypes.any,
+        /* Custom style override for the list item */
+        itemStyle: PropTypes.object,
+        /* Controls visibility of hover effect for the list item */
+        showHover: PropTypes.bool,
+    };
+
+    state = {
+        selectedCollaborator: null,
+    };
+
+    renderTimestampMessage = (interactedAt, interactionType) => {
+        const { intl } = this.props;
+        const lastActionMessage = determineInteractionMessage(interactionType, interactedAt);
+        const timeAgo = intl.formatRelativeTime
+            ? intl.formatRelativeTime(interactedAt - Date.now())
+            : intl.formatRelative(interactedAt);
+
+        if (lastActionMessage) {
+            return (
+                <FormattedMessage
+                    {...lastActionMessage}
+                    values={{
+                        timeAgo,
+                    }}
+                />
+            );
+        }
+        return null;
+    };
+
+    render() {
+        const { avatarStyle, collaborators, itemStyle, showHover } = this.props;
+        const { selectedCollaborator } = this.state;
+
+        return collaborators.map(collaborator => {
```

Comment:
Would it be more appropriate to create a `PresenceCollaborator` component that can be reused with or without an iterator?

Reply:
Updated!